pkg/pgxconn/implement.go

```diff
@@ -0,0 +1,105 @@
+package pgxconn
+
+import (
+	"time"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgproto3/v2"
+	"github.com/jackc/pgx/v4"
+)
+
+// rowWithMetric wraps the row returned by QueryRow() for metrics telemetry.
+type rowWithMetric struct {
+	r pgx.Row
+}
+
+func (w rowWithMetric) Scan(dest ...interface{}) error {
+	err := w.r.Scan(dest...)
+	if err != nil && err != pgx.ErrNoRows {
+		errorsTotal.With(promMethodLabel("query_row")).Inc()
+	}
+	return err
+}
+
+// rowsWithDuration wraps the rows and records the duration of the batch till it closed.
+type rowsWithDuration struct {
+	r     pgx.Rows
```

Comment:
```suggestion
	rows     pgx.Rows
```
nit: Let's use more descriptive name as the name is used in other functions. Otherwise we end up with lines like:
```golang
func (w rowsWithDuration) Err() error {
	return w.r.Err()
}
```
which force to look what `r` is.